### PR TITLE
Hide All context menu in workspace when displaying node context menu

### PR DESCRIPTION
**Pull request by QilongTang**
Please Note:
1. Before submitting the PR, please review [How to Contribute to Dynamo](https://github.com/DynamoDS/Dynamo/blob/master/CONTRIBUTING.md)
2. Dynamo Team will meet 1x a month to review PRs found on Github (Issues will be handled separately)
3. PRs will be reviewed from oldest to newest
4. If a reviewed PR requires changes by the owner, the owner of the PR has 30 days to respond. If the PR has seen no activity by the next session, it will be either closed by the team or depending on its utility will be taken over by someone on the team
5. PRs should use either Dynamo's default PR template or [one of these other template options](https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template) in order to be considered for review.
6. PRs that do not have one of the Dynamo PR templates completely filled out with all declarations satisfied will not be reviewed by the Dynamo team.
7. PRs made to the `DynamoRevit` repo will need to be cherry-picked into all the DynamoRevit Release branches that Dynamo supports. Contributors will be responsible for cherry-picking their reviewed commits to the other branches after a `LGTM` label is added to the PR.

### Purpose

Per https://jira.autodesk.com/browse/DYN-4509, https://jira.autodesk.com/browse/DYN-4459 and https://github.com/DynamoDS/Dynamo/issues/12532, fixing the issue that other popups are not hidden when displaying the node context menu. This fact will create all kinds of weird interactions.

1. Hide All other popups in workspace when displaying node context menu
2. Even if it fail, `UpdateModelValueImp` should not crash Dynamo. Catching the exception and logging it to Dynamo console

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

Hide All context menu in workspace when displaying node context menu

### Reviewers

@reddyashish @zeusongit 

### FYIs

@johnpierson

### CODE REVIEW

**Review by zeusongit — APPROVED** [review]

File: test/DynamoCoreTests/CoreTests.cs [hunk 1]
```diff
@@ -431,15 +431,17 @@ public void UpdateModelValue_MissingNode_ThrowsException()
...
         }
 
         [Test]
         [Category("UnitTests")]
         public void UpdateModelValue_EmptyList_ThrowsException()
         {
             var command = new DynCmd.UpdateModelValueCommand(Guid.Empty, new Guid[] { }, "", "");
-            Assert.Throws<ArgumentNullException>(() => CurrentDynamoModel.ExecuteCommand(command));
+            Assert.Throws<ArgumentNullException>(() => CurrentDynamoModel.CurrentWorkspace.UpdateModelValue(command.ModelGuids,
+                    command.Name, command.Value));
```
Comment (QilongTang) [inline test/DynamoCoreTests/CoreTests.cs:444]: @aparajit-pratap Let me know if you have a strong opinion on this. In my opinion, this should not crash Dynamo but currently, it is

File: src/DynamoCore/Models/DynamoModelCommands.cs [hunk 2]
```diff
@@ -597,12 +597,15 @@ private void UpdateModelValueImpl(UpdateModelValueCommand command)
...
-            if (targetWorkspace != null)
+            try
             {
-                targetWorkspace.UpdateModelValue(command.ModelGuids,
+                targetWorkspace?.UpdateModelValue(command.ModelGuids,
                     command.Name, command.Value);
             }
+            catch (Exception ex)
+            {
+                Logger.LogError(ex.Message);
```
Comment (aparajit-pratap) [inline src/DynamoCore/Models/DynamoModelCommands.cs:607]: Do we know what kinds of exceptions are these and why are they thrown?

Comment (QilongTang) [reply src/DynamoCore/Models/DynamoModelCommands.cs:607]: There are code inside the function throw the exception, but they are not caught anywhere, so Dynamo will crash

File: src/DynamoCoreWpf/Views/Core/WorkspaceView.xaml.cs [hunk 3]
```diff
@@ -253,15 +255,23 @@ private void ShowHidePopup(ShowHideFlags flag, Popup popup)
...
             {
                 ShowHideContextMenu(ShowHideFlags.Hide);
                 ShowHideInCanvasControl(ShowHideFlags.Hide);
             }
+            // If triggered on node level, make sure node popups are also hidden
+            if(sender is NodeView && (PortContextMenu.IsOpen || NodeAutoCompleteSearchBar.IsOpen) )
+            {
+                ShowHidePopup(ShowHideFlags.Hide, PortContextMenu);
+                ShowHidePopup(ShowHideFlags.Hide, NodeAutoCompleteSearchBar);
+            }
```
Comment (aparajit-pratap) [inline src/DynamoCoreWpf/Views/Core/WorkspaceView.xaml.cs:274]: So if the sender is not the nodeview (if it's the workspace), the node level popups can still be open?

Comment (QilongTang) [reply src/DynamoCoreWpf/Views/Core/WorkspaceView.xaml.cs:274]: Yes, this is the intended design from UX originally

File: test/DynamoCoreTests/CoreTests.cs (hunk 1 above)
Comment (aparajit-pratap) [reply test/DynamoCoreTests/CoreTests.cs:444]: Can you remind me how these exceptions can be reproduced exactly? Looking at that old PR, I see that these exceptions are thrown when either the nodemodel is not found in the workspace or the list of nodemodels passed is null or empty, but how exactly can these situations be reproduced.

**Review by aparajit-pratap — COMMENTED** [review]
I think overall it's better to catch unhandled exceptions that would otherwise lead to a crash. Just one question as to steps that a user can follow in Dynamo that would lead to these crashes earlier?

**Comment by QilongTang** [comment]
> I think overall it's better to catch unhandled exceptions that would otherwise lead to a crash. Just one question as to steps that a user would follow in Dynamo that would lead to these crashes earlier?

Yes, @aparajit-pratap You can refer to https://github.com/DynamoDS/Dynamo/issues/12532, although it should be avoided by my PR but I think catching the exception would be better than just leaving it as it is. Let me know

**Review by aparajit-pratap — APPROVED** [review]

**Pull request merged**